barretenberg/cpp/src/barretenberg/ecc/scalar_multiplication/scalar_multiplication.test.cpp

```diff
@@ -479,6 +481,86 @@ TYPED_TEST(ScalarMultiplicationTest, MSMEmptyPolynomial)
 
+// Helper function to generate scalars with specified sparsity
+template <typename ScalarField>
+std::vector<ScalarField> generate_sparse_scalars(size_t num_scalars, double sparsity_rate, auto& rng)
+{
+    std::vector<ScalarField> scalars(num_scalars);
+    for (size_t i = 0; i < num_scalars; ++i) {
+        // Generate random value to determine if this scalar should be zero
+        double rand_val = static_cast<double>(rng.get_random_uint32()) / static_cast<double>(UINT32_MAX);
+        if (rand_val < sparsity_rate) {
+            scalars[i] = 0;
+        } else {
+            scalars[i] = ScalarField::random_element(&rng);
+        }
+    }
+    return scalars;
+}
+
+// Test different MSM strategies with detailed benchmarking
+// NOTE this requres BB_BENCH=1 to be set before the test command
+TYPED_TEST(ScalarMultiplicationTest, BenchBatchMsm)
+{
+#ifndef __wasm__
+    if (!bb::detail::use_bb_bench) {
+#else
+    {
+#endif
+        std::cout
+            << "Skipping BatchMultiScalarMulStrategyComparison as BB_BENCH=1 is not passed (OR we are in wasm).\n";
+        return;
+    }
+    SCALAR_MULTIPLICATION_TYPE_ALIASES
+
+    using AffineElement = typename Curve::AffineElement;
+
+    const size_t num_msms = 3;
+    const size_t msm_max_size = 1 << 17;
+    const double max_sparsity = 0.1;
+
+    // Generate test data with varying sparsity
+    std::vector<std::span<const AffineElement>> all_points;
+    std::vector<std::span<ScalarField>> all_scalars;
+    std::vector<std::vector<ScalarField>> scalar_storage;
+
+    for (size_t i = 0; i < num_msms; ++i) {
+        // Generate random sizes and density of 0s
+        const size_t size = engine.get_random_uint64() % msm_max_size;
+        const double sparsity = engine.get_random_uint8() / 255.0 * max_sparsity;
+        auto scalars = generate_sparse_scalars<ScalarField>(size, sparsity, engine);
+        scalar_storage.push_back(std::move(scalars));
+
+        std::span<const AffineElement> points(&TestFixture::generators[i], size);
+        all_points.push_back(points);
+        all_scalars.push_back(scalar_storage.back());
+    }
+    auto func = [&]<bb::detail::OperationLabel thread_prefix>(size_t num_threads) {
+        set_hardware_concurrency(num_threads);
+        // Strategy 1: Individual MSMs
+        {
+            BB_BENCH_NAME((bb::detail::concat<thread_prefix, "IndividualMSMs">()));
+            for (size_t i = 0; i < num_msms; ++i) {
+                std::vector<std::span<const AffineElement>> single_points = { all_points[i] };
+                std::vector<std::span<ScalarField>> single_scalars = { all_scalars[i] };
+                scalar_multiplication::MSM<Curve>::batch_multi_scalar_mul(single_points, single_scalars);
+            }
+        }
+        // Strategy 4: Full Batch
+        {
+            BB_BENCH_NAME((bb::detail::concat<thread_prefix, "BatchMSMs">()));
+            scalar_multiplication::MSM<Curve>::batch_multi_scalar_mul(all_points, all_scalars);
+        }
```

Comment:
couldn't hurt to add a little sanity check here to see if the results are equal (to give confidence the bench is real)